Sources/Controllers/Panels/OAMapPanelViewController.mm

```diff
@@ -2568,27 +2568,32 @@ - (void)doShowGpxItem:(OAGPX *)item
     if (_dashboard)
         [self closeDashboard];
 
-    OAMapRendererView *renderView = (OAMapRendererView *) _mapViewController.view;
     OATargetPoint *targetPoint = [[OATargetPoint alloc] init];
-    if (item.bounds.center.latitude == DBL_MAX)
+    if (!state || !CLLocationCoordinate2DIsValid(state.pinLocation))
     {
-        OsmAnd::LatLon latLon = OsmAnd::Utilities::convert31ToLatLon(renderView.target31);
-        targetPoint.location = CLLocationCoordinate2DMake(latLon.latitude, latLon.longitude);
-        _targetLatitude = latLon.latitude;
-        _targetLongitude = latLon.longitude;
+        if (CLLocationCoordinate2DIsValid(item.bounds.center))
+        {
+            targetPoint.location = CLLocationCoordinate2DMake(item.bounds.center.latitude, item.bounds.center.longitude);
```

Comment:
item.bounds.center is of CLLocationCoordinate2D type, no need to create a new coordinate here